[IDE] Wrap the Value getter in a try catch block

iainx opened the pull request:
Wrap the value getter in a try catch block to prevent the crash in 706892. Log the exception in case
we reproduce it.

Fixes VSTS #706892

mrward commented on main/src/core/MonoDevelop.Ide/MonoDevelop.Components.PropertyGrid/PropertyEditorCell.cs:
```diff
@@ -106,18 +107,25 @@ string GetNormalizedText (string s)
+				try {
+					return Instance != null ? Property.GetValue (Instance) : null;
+				} catch (Exception e) {
+					LoggingService.LogInternalError ("Hit Bug 706892", e);
```
Should we log more information here such as the Instance's full type name? I am assuming the exception may not include this information about the type itself.

iainx replied:
Hmm, you're right. The exception I saw triggered did include that information, but it would be good to have it just in case different exceptions can be hit

Commit:
[IDE] Wrap the PropertyCell Value getter in a try catch block

There is a crash in the PropertyCell value getter, which this try catch will
work around

Fixes VSTS #706892
